src/ameba/rule/style/verbose_block.cr

```diff
@@ -0,0 +1,229 @@
+    end
+
+    private PREFIX_OPERATORS = {"+", "-"}
+    private OPERATOR_CHARS   =
+      {'[', ']', '!', '=', '>', '<', '~', '+', '-', '*', '/', '%', '^', '|', '&'}
+
+    protected def prefix_operator?(node)
+      node.name.in?(PREFIX_OPERATORS) && node.args.empty?
+    end
+
+    protected def operator?(name)
+      name.each_char do |char|
+        return false unless char.in?(OPERATOR_CHARS)
+      end
+      !name.empty?
+    end
+
+    protected def setter?(name)
+      !name.empty? && name[0].letter? && name.ends_with?('=')
+    end
+
+    protected def valid_length?(code)
+      if max_length = self.max_length
+        return code.size <= max_length
+      end
+      true
+    end
+
+    protected def valid_line_length?(node, code)
+      if max_line_length = self.max_line_length
+        if location = node.name_location
+          final_line_length = location.column_number + code.size
+          return final_line_length <= max_line_length
+        end
+      end
+      true
+    end
+
+    protected def reference_count(node, obj : Crystal::Var)
+      i = 0
+      case node
+      when Crystal::Call
+        i += reference_count(node.obj, obj)
+        i += reference_count(node.block, obj)
+
+        node.args.each do |arg|
+          i += reference_count(arg, obj)
+        end
+        node.named_args.try &.each do |arg|
+          i += reference_count(arg.value, obj)
+        end
+      when Crystal::Block
+        i += reference_count(node.body, obj)
+      when Crystal::Var
+        i += 1 if node == obj
+      end
+      i
+    end
+
+    protected def args_to_s(io : IO, node : Crystal::Call, skip_last_arg = false)
+      node.args.dup.tap do |args|
+        args.pop? if skip_last_arg
+        args.join io, ", "
+        node.named_args.try do |named_args|
+          io << ", " unless args.empty? || named_args.empty?
+          named_args.join io, ", " do |arg, inner_io|
+            inner_io << arg.name << ": " << arg.value
+          end
+        end
+      end
+    end
+
+    protected def args_to_s(node : Crystal::Call, skip_last_arg = false)
+      String.build { |io| args_to_s(io, node, skip_last_arg) }
+    end
+
+    protected def node_to_s(node : Crystal::Call)
+      String.build do |str|
+        case name = node.name
+        when "[]"
+          str << '['
+          args_to_s(str, node)
+          str << ']'
+        when "[]?"
+          str << '['
+          args_to_s(str, node)
+          str << "]?"
+        when "[]="
+          str << '['
+          args_to_s(str, node, skip_last_arg: true)
+          str << "]=(" << node.args.last? << ')'
+        else
+          str << name
+          if !node.args.empty? || (node.named_args && !node.named_args.try(&.empty?))
+            str << '('
+            args_to_s(str, node)
+            str << ')'
+          end
+          str << " {...}" if node.block
+        end
+      end
+    end
+
+    protected def call_code(call, body)
```

Comment:
I think it would be great to give it just a bit more meaningful name and extract this to the utility class (or mixin) to make it reusable in the future.

Reply:
Fair point, any suggestions onto where should it go?

Reply:
I would put it to [ast/util](https://github.com/crystal-ameba/ameba/blob/feature/verbose-block-rule/src/ameba/ast/util.cr) class. However, if you feel uncomfortable with it, it can be done later on demand.

Reply:
Cool, I'd prefer to do it in the followup PR though.